Feature 05

JoeyAlanS opened the pull request.

JoeyAlanS committed:
###initial commit

JoeyAlanS committed:
###add
-Added idea

JoeyAlanS committed:
###initial commit

JoeyAlanS committed:
Merge branch 'Feature_part_5' of https://github.com/JoeyAlanS/Integration-project-backend into Feature_part_5

# Conflicts:
#	.idea/workspace.xml
#	src/test/java/com/eletra/controllers/CategoryControllerTest.java
#	src/test/java/com/eletra/controllers/LineupControllerTest.java
#	src/test/java/com/eletra/controllers/ModelControllerTest.java
#	src/test/java/com/eletra/repositories/CategoryRepositoryTest.java
#	src/test/java/com/eletra/repositories/LineupRepositoryTest.java
#	src/test/java/com/eletra/repositories/ModelRepositoryTest.java

JoeyAlanS committed:
###Initial commit

JoeyAlanS committed:
###Added
- New classes of test

JoeyAlanS committed:
###initial commit

JoeyAlanS committed:
###initial commit

JoeyAlanS committed:
###initial commit

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/CategoryControllerTest.java:
por favor, adicionar os modificadores nos testes

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/CategoryControllerTest.java:
não foi testado o caso em que o categoryEntity seja diferente de nulo no método deleteCategoryEntity

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/CategoryControllerTest.java:
```diff
@@ -0,0 +1,111 @@
+        List<CategoryEntity> mockCategoryList = new ArrayList<>();
+        mockCategoryList.add(new CategoryEntity("Category1", (short) 1));
+        mockCategoryList.add(new CategoryEntity("Category2", (short) 2));
+        LineupEntity line = new LineupEntity("Line", (short) 1);
```
essa linha não é usada, favor, apagá-la

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/CategoryControllerTest.java:
```diff
@@ -0,0 +1,111 @@
+
+    @Test
+    void deleteCategoryEntityTest() {
+        CategoryEntity mockCategory = new CategoryEntity("Category1", (short) 1);
```
essa linha não é usada, favor, apagá-la

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/LineupControllerTest.java:
não foi testado o caso em que o lineEntity seja igual a nulo no método deleteLineupEntity

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/LineupControllerTest.java:
por favor, adicionar os modificadores nos testes

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/ModelControllerTest.java:
por favor, adicionar os modificadores nos testes

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/ModelControllerTest.java:
não foi testado o caso em que o modelEntity seja igual a nulo no método deleteModelEntity

gustavo-lopes-dev commented on src/test/java/com/eletra/repositories/CategoryRepositoryTest.java:
Os testes feitos nessa classe não são testes da CategoryRepository e sim da CategoryService. Portanto, por favor, mova esse teste para o pacote services e refatore o nome da classe

gustavo-lopes-dev commented on src/test/java/com/eletra/repositories/LineupRepositoryTest.java:
Os testes feitos nessa classe não são testes da LineupRepository e sim da LineupService. Portanto, por favor, mova esse teste para o pacote services e refatore o nome da classe

gustavo-lopes-dev commented on src/test/java/com/eletra/repositories/ModelRepositoryTest.java:
Os testes feitos nessa classe não são testes da ModelRepository e sim da ModelService. Portanto, por favor, mova esse teste para o pacote services e refatore o nome da classe

gustavo-lopes-dev reviewed (changes requested):
Foi realizado o PR e foram identificados pontos de melhoria e/ou correção

JoeyAlanS committed:
###Changed
- changed package name of repositories to services
- added 3 new classes of check

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/CategoryControllerTest.java:
```diff
@@ -82,9 +78,27 @@ void postCategoryEntityTest() {
+    public void deleteCategoryEntityTest() {
         CategoryEntity mockCategory = new CategoryEntity("Category1", (short) 1);
 
+        when(mockCategoryRepository.findByCategoryName("Category1"))
+                .thenReturn(mockCategory)
+                .thenReturn(null);
```
Corrigir mock, está sendo realizado dois "thenReturn"

JoeyAlanS replied:
Corrigido

gustavo-lopes-dev commented on src/test/java/com/eletra/controllers/LineupControllerTest.java:
```diff
@@ -72,10 +73,23 @@ void postLineupEntityTest() {
         LineupEntity mockLine = new LineupEntity("Line1", (short) 1);
 
-        when(mockLineupRepository.findByLineName("Line1")).thenReturn(mockLine).thenReturn(null);
+        when(mockLineupRepository.findByLineName("Line1"))
+                .thenReturn(mockLine)
+                .thenReturn(null);
```
Corrigir mock, está sendo realizado dois "thenReturn"

JoeyAlanS replied:
Corrigido

gustavo-lopes-dev reviewed (changes requested):
Foram encontrados pontos para correção

gustavo-lopes-dev commented:
Por favor, também não esquecer de colocar a anotação After nas classes de testes. Ele é fundamental pra limpar as instâncias criadas na anotação Before e evitar problemas com memória

JoeyAlanS committed:
###Changed
- Changed methods on CategoryController
- Changed methods on LineupController
- Changed methods on ModelController

JoeyAlanS committed:
###Added
- Added notations

Bacuriim reviewed (approved):
Sem mais pontos a corrigir, Aprovado!

JoeyAlanS committed:
###Changed
- Changeg name linupy to lineup
